src/ds5/ds5-auto-calibration.cpp

```diff
@@ -294,84 +321,166 @@ namespace librealsense
+            }
 
             // Begin auto-calibration
-            _hw_monitor->send(command{ ds::AUTO_CALIB, py_rx_calib_begin, speed, 0, param.param_4 });
+            if (host_assistance == 0 || host_assistance == 1)
+                _hw_monitor->send(command{ ds::AUTO_CALIB, py_rx_calib_begin, speed, 0, p4 });
+            
+            if (host_assistance != 1)
+            {
+                if (host_assistance == 2)
+                {
+                    command cmd(ds::AUTO_CALIB, interactive_scan_control, 0, 0);
+                    uint8_t* p = reinterpret_cast<uint8_t*>(&step_count_v3);
+                    cmd.data.push_back(p[0]);
+                    cmd.data.push_back(p[1]);
+                    for (uint16_t i = 0; i < step_count_v3; ++i)
+                    {
+                        p = reinterpret_cast<uint8_t*>(fill_factor + i);
+                        cmd.data.push_back(p[0]);
+                        cmd.data.push_back(p[1]);
+                    }
 
-            DirectSearchCalibrationResult result{};
+                    _hw_monitor->send(cmd);
+                }
 
-            int count = 0;
-            int retries = 0;
-            bool done = false;
+                DirectSearchCalibrationResult result{};
 
-            auto start = std::chrono::high_resolution_clock::now();
-            auto now = start;
+                int count = 0;
+                int retries = 0;
+                bool done = false;
 
-            // While not ready...
-            do
-            {
-                std::this_thread::sleep_for(std::chrono::milliseconds(200));
+                auto start = std::chrono::high_resolution_clock::now();
+                auto now = start;
 
-                // Check calibration status
-                try
+                // While not ready...
+                do
                 {
-                    auto res = _hw_monitor->send(command{ ds::AUTO_CALIB, py_rx_calib_check_status });
+                    std::this_thread::sleep_for(std::chrono::milliseconds(200));
 
-                    if (res.size() < sizeof(DirectSearchCalibrationResult))
-                        throw std::runtime_error("Not enough data from CALIB_STATUS!");
+                    // Check calibration status
+                    try
+                    {
+                        auto res = _hw_monitor->send(command{ ds::AUTO_CALIB, py_rx_calib_check_status });
 
-                    result = *reinterpret_cast<DirectSearchCalibrationResult*>(res.data());
-                    done = result.status != RS2_DSC_STATUS_RESULT_NOT_READY;
-                }
-                catch (const std::exception& ex)
-                {
-                    if (!((retries++) % 5)) // Add log debug once in a sec
+                        if (res.size() < sizeof(DirectSearchCalibrationResult))
+                            throw std::runtime_error("Not enough data from CALIB_STATUS!");
```

Comment:
Replace throws with branching. We don't want to abuse the error handling for flow control

Reply:
Done.